datasets/covid19_italy/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,149 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# CSV transform for: covid19_italy
+
+import datetime
+import json
+import logging
+import os
+import pathlib
+import typing
+
+import pandas as pd
+import requests
+from google.cloud import storage
+
+
+def main(
+    source_url: str,
+    source_file: pathlib.Path,
+    target_file: pathlib.Path,
+    target_gcs_bucket: str,
+    target_gcs_path: str,
+    headers: typing.List[str],
+    rename_mappings: dict,
+    pipeline_name: str,
+):
+
+    logging.info(
+        "Covid-19 Italy process started at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+    logging.info("creating 'files' folder")
+    pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+
+    logging.info(f"Downloading file {source_url}")
+    download_file(source_url, source_file)
+
+    # open the input file
+    logging.info(f"Opening file {source_file}")
+
+    df = pd.read_csv(str(source_file))
+
+    # steps in the pipeline
+    logging.info(f"Transformation Process Starting.. {source_file}")
+
+    # rename the headers
+    logging.info(f"Transform: Renaming Headers.. {source_file}")
+    rename_headers(df, rename_mappings)
+
+    # create location_geom
+    logging.info(f"Transform: Creating Geometry Column.. {pipeline_name}")
+    if pipeline_name == "data_by_province" or pipeline_name == "data_by_region":
+        df["location_geom"] = (
+            "POINT("
+            + df["longitude"].astype(str).replace("nan", "")
+            + " "
+            + df["latitude"].astype(str).replace("nan", "")
+            + ")"
+        )
+        # replace blank POINT( ) valye with blank
+        df.location_geom = df.location_geom.replace("POINT( )", "")
+
+    # reorder headers in output
+    logging.info("Transform: Reordering headers..")
+    df = df[headers]
+
+    # steps in the pipeline
+    logging.info(f"Transformation Process complete .. {source_file}")
+
+    # save to output file
+    logging.info(f"Saving to output file.. {target_file}")
+
+    try:
+        # save_to_new_file(df, file_path=str(target_file))
+        save_to_new_file(df, file_path=str(target_file))
+    except Exception as e:
+        logging.error(f"Error saving output file: {e}.")
+
+    logging.info(
+        f"Uploading output file to.. gs://{target_gcs_bucket}/{target_gcs_path}"
+    )
+    upload_file_to_gcs(target_file, target_gcs_bucket, target_gcs_path)
+
+    # log completion
+    logging.info(
+        "Covid-19 Italy process completed at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+
+def convert_dt_format(date_str, time_str):
```

Comment:
Also add the types for `date_str` and `time_str`
```
convert_dt_format(date_str: str, time_str: str)
```
Please do this for other functions below as well.

Reply:
Done.